src/client/sandbox/native-methods.js

```diff
@@ -41,11 +39,12 @@ class NativeMethods {
         this.documentCreateTouchList     = doc.createTouchList || docProto.createTouchList;
 
         // getters/setters
-        const docPrototype             = win.Document.prototype;
-        const htmlDocPrototype         = win.HTMLDocument.prototype;
-        const documentCookieDescriptor = win.Object.getOwnPropertyDescriptor(isFirefox ? htmlDocPrototype : docPrototype, 'cookie');
+        const docPrototype = win.Document.prototype;
+
+        this.documentAllPropOwnerName    = docPrototype.hasOwnProperty('all') ? 'Document' : 'HTMLDocument';
+        this.documentCookiePropOwnerName = docPrototype.hasOwnProperty('cookie') ? 'Document' : 'HTMLDocument';
```

Comment:
Move this code `docPrototype.hasOwnProperty('all') ? 'Document' : 'HTMLDocument';` 
to the instance level utils function `NativeMethods.getPropOwnerName (prop).
Store `docProtorype` in the `nativeMethods` if it's necessary.